Source/Core/Common/FixedSizeQueue.h

```diff
@@ -45,6 +42,46 @@ class FixedSizeQueue
     tail = (tail + 1) % N;
   }
 
+  // Copies over an array, loops over if num is greater than max_size
+  void push_array(const T* t, size_t num)
```

Comment:
To be honest, I'm not convinced this complicated logic is required. Do you have profiling data that shows that calling `emplace()` in loop is a bottleneck?

Reply:
I don't have a profiling no, and this is definitely not mandatory to have, but once implemented it's there, so, why not?
I made this for the DPLII which given the new sample rates implementations in my other draft audio PR, could end up copying 2688000 floats per second, so I though a memcopy would have been better.

Reply:
Because it's possibly a premature optimisation and as the other review comments show it's easy to miss edge cases like T not being trivially copyable.

If this is not even a bottleneck, it's a lot of code that does not have to exist and be maintained.

Reply:
Over the max number of floats DPLII can copy per audio frame, this is from ~17% to ~40% faster in release on MSVC v14.28.
I've tested a lot of times with microseconds accuracy. The difference goes bigger as the copy num grows of course.